packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -214,16 +214,20 @@ export class ChromeService {
       return `kbnVersion-${formattedVersionClass}`;
     };
 
+    const chromeStyle$ = chromeStyleSubject$.pipe(
+      filter((style): style is ChromeStyle => style !== undefined),
+      takeUntil(this.stop$)
+    );
     const setChromeStyle = (style: ChromeStyle) => {
-      if (style === chromeStyle$.getValue()) return;
-      chromeStyle$.next(style);
+      if (style === chromeStyleSubject$.getValue()) return;
+      chromeStyleSubject$.next(style);
     };
 
     const headerBanner$ = new BehaviorSubject<ChromeUserBanner | undefined>(undefined);
     const bodyClasses$ = combineLatest([
       headerBanner$,
       this.isVisible$!,
-      chromeStyle$,
+      chromeStyleSubject$,
```

Comment:
I think here we can keep `chromeStyle$`

Reply:
Same as above, the filtered observable will emit less often, so the `combineLatest` might also emit less often.

Reply:
This one broke, because it didn't emit at all under certain circumstances. So I reverted back to using the unfiltered subject. Is that ok with you?

Reply:
> it didn't emit at all under certain circumstances

That worries me a bit. It would be great to know under which circumstance it does not emit a value as if it doesn't basically Kibana is broken (no header). Do you remember the steps to reproduce a moment where it didn't emit a value?

Reply:
Ah maybe bodyClasses needs to resolve first before moving forward and render Kibana, which thens render the navigation plugin which sets the Chrome style. 😊

Reply:
yes, that is my hypothesis as well